### PR TITLE
[tests-only][full-ci] Suppress NoSuchElementError on isVisible whenever expected not to be visible

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
suppress the NoSuchElementError when the element is expected not to be visible or if there is conditional actions

## Related Issue
- Fixes (https://github.com/owncloud/web/issues/6463)

## Motivation and Context
Clean log

## How Has This Been Tested?
- test environment:

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles4</strong> https://drone.owncloud.com/owncloud/web/23253/58/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIRenameFolders-renameFolders_feature-L79.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles3</strong> https://drone.owncloud.com/owncloud/web/23253/57/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIRenameFiles-renameFiles_feature-L246.png
  

### webUIRenameFiles-renameFiles_feature-L90.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingAndTrashbin</strong> https://drone.owncloud.com/owncloud/web/23253/33/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUITrashbinRestore-trashbinRestore_feature-L14.png
  

### webUITrashbinRestore-trashbinRestore_feature-L30.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10FileOperations</strong> https://drone.owncloud.com/owncloud/web/23253/14/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIDeleteFilesFolders-deleteFilesFolders_feature-L125.png
  

### webUIDeleteFilesFolders-deleteFilesFolders_feature-L236.png
  

### webUIDeleteFilesFolders-deleteFilesFolders_feature-L253.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingIntGroupsToRoot</strong> https://drone.owncloud.com/owncloud/web/23253/28/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingInternalGroupsToRoot-shareWithGroups_feature-L69.png
  

### webUISharingInternalGroupsToRoot-shareWithGroups_feature-L70.png
  

### webUISharingInternalGroupsToRoot-shareWithGroups_feature-L71.png
  

### webUISharingInternalGroupsToRootEdgeCases-shareWithGroupsEdgeCases_feature-L36.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingIntUsers2</strong> https://drone.owncloud.com/owncloud/web/23253/30/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingInternalUsersToRoot-shareWithUsers_feature-L157.png
  

### webUISharingInternalUsersToRoot-shareWithUsers_feature-L252.png
  

### webUISharingInternalUsersToRoot-shareWithUsers_feature-L253.png
  

### webUISharingInternalUsersToRoot-shareWithUsers_feature-L254.png
  

### webUISharingInternalUsersToRoot-shareWithUsers_feature-L48.png
  

### webUISharingInternalUsersToRoot-shareWithUsers_feature-L49.png
  

### webUISharingInternalUsersToRoot-shareWithUsers_feature-L50.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp2</strong> https://drone.owncloud.com/owncloud/web/23253/72/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingInternalUsers-shareWithUsers_feature-L52.png
  

### webUISharingInternalUsers-shareWithUsers_feature-L53.png
  

### webUISharingInternalUsers-shareWithUsers_feature-L54.png
  

### webUISharingInternalUsersToRoot-shareWithUsers_feature-L48.png
  

### webUISharingInternalUsersToRoot-shareWithUsers_feature-L49.png
  

### webUISharingInternalUsersToRoot-shareWithUsers_feature-L50.png
  

### webUITrashbinRestore-trashbinRestore_feature-L14.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternal3</strong> https://drone.owncloud.com/owncloud/web/23253/61/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIResharing1-reshareUsers_feature-L198.png
  

### webUIResharing1-reshareUsers_feature-L214.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10Files5</strong> https://drone.owncloud.com/owncloud/web/23253/20/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIRenameFolders-renameFolders_feature-L79.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp1</strong> https://drone.owncloud.com/owncloud/web/23253/71/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIFavorites-favoritesFile_feature-L14.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles1</strong> https://drone.owncloud.com/owncloud/web/23253/55/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIDeleteFilesFolders-deleteFilesFolders_feature-L125.png
  

### webUIDeleteFilesFolders-deleteFilesFolders_feature-L236.png
  

### webUIDeleteFilesFolders-deleteFilesFolders_feature-L253.png
  

### webUIFilesCopy-copy_feature-L70.png
  

### webUIFilesCopy-copy_feature-L71.png
  

### webUIFilesCopy-copy_feature-L72.png
  

### webUIFilesCopy-copy_feature-L73.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingIntUsers1</strong> https://drone.owncloud.com/owncloud/web/23253/29/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingInternalUsers-shareWithUsers_feature-L158.png
  

### webUISharingInternalUsers-shareWithUsers_feature-L275.png
  

### webUISharingInternalUsers-shareWithUsers_feature-L276.png
  

### webUISharingInternalUsers-shareWithUsers_feature-L277.png
  

### webUISharingInternalUsers-shareWithUsers_feature-L52.png
  

### webUISharingInternalUsers-shareWithUsers_feature-L53.png
  

### webUISharingInternalUsers-shareWithUsers_feature-L54.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10Files3</strong> https://drone.owncloud.com/owncloud/web/23253/18/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIRenameFiles-renameFiles_feature-L199.png
  

### webUIRenameFiles-renameFiles_feature-L224.png
  

### webUIRenameFiles-renameFiles_feature-L246.png
  

### webUIRenameFiles-renameFiles_feature-L90.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10Sharing2</strong> https://drone.owncloud.com/owncloud/web/23253/24/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIResharingToRoot-reshareUsers_feature-L160.png
  

### webUIResharingToRoot-reshareUsers_feature-L204.png
  

### webUIResharingToRoot-reshareUsers_feature-L216.png
  

### webUIResharingToRoot-reshareUsers_feature-L228.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10Resharing</strong> https://drone.owncloud.com/owncloud/web/23253/22/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIResharing1-reshareUsers_feature-L145.png
  

### webUIResharing1-reshareUsers_feature-L198.png
  

### webUIResharing1-reshareUsers_feature-L214.png
  

### webUIResharing1-reshareUsers_feature-L230.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternal2</strong> https://drone.owncloud.com/owncloud/web/23253/60/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingInternalUsers-shareWithUsers_feature-L158.png
  

### webUISharingInternalUsers-shareWithUsers_feature-L243.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10Files2</strong> https://drone.owncloud.com/owncloud/web/23253/17/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIFilesActionMenu-fileFolderActionMenu_feature-L14.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles2</strong> https://drone.owncloud.com/owncloud/web/23253/56/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIFilesActionMenu-fileFolderActionMenu_feature-L14.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10Move</strong> https://drone.owncloud.com/owncloud/web/23253/21/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIMoveFilesFolders-moveFiles_feature-L79.png
  

### webUIMoveFilesFolders-moveFiles_feature-L81.png
  

### webUIMoveFilesFolders-moveFiles_feature-L82.png
  

### webUIMoveFilesFolders-moveFolders_feature-L66.png
  

### webUIMoveFilesFolders-moveFolders_feature-L68.png
  

### webUIMoveFilesFolders-moveFolders_feature-L69.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingAccept</strong> https://drone.owncloud.com/owncloud/web/23253/15/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingAcceptSharesToRoot-acceptShares_feature-L108.png
  


</p></details>

File: tests/acceptance/features/webUIDeleteFilesFolders/deleteFilesFolders.feature [hunk 1]
```diff
@@ -240,7 +240,7 @@ Feature: deleting files and folders
     And user "Brian" has created file "lorem.txt" in the server
     And user "Brian" has shared folder "simple-folder" with user "Alice" in the server
     And user "Brian" has shared file "lorem.txt" with user "Alice" in the server
-    And the user has browsed to the shared-with-me page
+    And the user has browserrored to the shared-with-me page
```
Comment (pascalwengerter) [inline tests/acceptance/features/webUIDeleteFilesFolders/deleteFilesFolders.feature:243]: This looks like a typo, no? @saw-jan

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternal2</strong> https://drone.owncloud.com/owncloud/web/23276/60/1
The following scenarios passed on retry:
- webUISharingInternalUsers/shareWithUsers.feature:342

**Comment by ownclouders** [comment]
Results for <strong>oCISTrashbinUploadMoveJourney</strong> https://drone.owncloud.com/owncloud/web/23276/69/1
The following scenarios passed on retry:
- webUITrashbinRestore/trashbinRestore.feature:46

**Comment by ownclouders** [comment]
Results for <strong>oC10Resharing</strong> https://drone.owncloud.com/owncloud/web/23276/22/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIResharing1-reshareUsers_feature-L145.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10Sharing2</strong> https://drone.owncloud.com/owncloud/web/23276/24/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIResharingToRoot-reshareUsers_feature-L160.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp1</strong> https://drone.owncloud.com/owncloud/web/23276/71/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIPrivateLinks-accessingPrivateLinks_feature-L9.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10Resharing</strong> https://drone.owncloud.com/owncloud/web/23313/22/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIResharing1-reshareUsers_feature-L145.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic2</strong> https://drone.owncloud.com/owncloud/web/23313/68/1
The following scenarios passed on retry:
- webUISharingPublicManagement/shareByPublicLink.feature:119

**Comment by ownclouders** [comment]
Results for <strong>oC10Sharing2</strong> https://drone.owncloud.com/owncloud/web/23313/24/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIResharingToRoot-reshareUsers_feature-L160.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingAndUpload</strong> https://drone.owncloud.com/owncloud/web/23334/66/1
The following scenarios passed on retry:
- webUIUpload/upload.feature:94

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp1</strong> https://drone.owncloud.com/owncloud/web/23334/71/1
The following scenarios passed on retry:
- webUIPrivateLinks/accessingPrivateLinks.feature:9

**Comment by ownclouders** [comment]
Results for <strong>oC10iPhone1</strong> https://drone.owncloud.com/owncloud/web/23338/48/1
The following scenarios passed on retry:
- webUIPrivateLinks/accessingPrivateLinks.feature:9

**Comment by ownclouders** [comment]
Results for <strong>oC10XGAPortrait1</strong> https://drone.owncloud.com/owncloud/web/23338/44/1
The following scenarios passed on retry:
- webUIPrivateLinks/accessingPrivateLinks.feature:9

**Comment by ownclouders** [comment]
Results for <strong>oCISTrashbinUploadMoveJourney</strong> https://drone.owncloud.com/owncloud/web/23338/69/1
The following scenarios passed on retry:
- webUITrashbinRestore/trashbinRestore.feature:84

File: tests/acceptance/pageObjects/webPage.js [hunk 2]
```diff
@@ -15,34 +15,43 @@ module.exports = {
...
-              this.click('#files-open-search-btn')
-                .waitForElementVisible('@searchInputFieldLowResolution')
-                .setValue('@searchInputFieldLowResolution', [searchTerm])
-                .click('@searchGlobalButton')
-            } else {
-              this.waitForElementVisible('@searchInputFieldHighResolution')
-                .setValue('@searchInputFieldHighResolution', [searchTerm])
-                .click('@searchGlobalButton')
+          .isVisible(
+            { selector: '#files-open-search-btn', suppressNotFoundErrors: true },
```
Comment (kiranparajuli589) [inline tests/acceptance/pageObjects/webPage.js:19]: use a variable for the selector

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternal2</strong> https://drone.owncloud.com/owncloud/web/23416/60/1
The following scenarios passed on retry:
- webUISharingInternalUsers/shareWithUsers.feature:342

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingAndUpload</strong> https://drone.owncloud.com/owncloud/web/23416/66/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUIUpload-upload_feature-L94.png
  


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp1</strong> https://drone.owncloud.com/owncloud/web/23416/71/1
The following scenarios passed on retry:
- webUIPrivateLinks/accessingPrivateLinks.feature:9

**Comment by kiranparajuli589** [comment]
ci failed with `Error response from daemon: No such image: owncloudci/php:7.4`

build restarted!

**Comment by ownclouders** [comment]
Results for <strong>oC10IntegrationApp1</strong> https://drone.owncloud.com/owncloud/web/23419/71/1
The following scenarios passed on retry:
- webUIPrivateLinks/accessingPrivateLinks.feature:17

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingSharingInd</strong> https://drone.owncloud.com/owncloud/web/23421/32/1
The following scenarios passed on retry:
- webUISharingInternalGroupsToRootSharingIndicator/shareWithGroups.feature:124

**Comment by ownclouders** [comment]
Results for <strong>oCISTrashbinUploadMoveJourney</strong> https://drone.owncloud.com/owncloud/web/23421/69/1
The following scenarios passed on retry:
- webUITrashbinRestore/trashbinRestore.feature:84

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic1</strong> https://drone.owncloud.com/owncloud/web/23421/67/1
The following scenarios passed on retry:
- webUISharingPublicBasic/publicLinkCreate.feature:190

File: tests/acceptance/pageObjects/FilesPageElement/SharingDialog/collaboratorsDialog.js [hunk 3]
```diff
@@ -57,18 +57,21 @@ module.exports = {
       this.useXpath().waitForElementVisible(editExpirationSelector).click(editExpirationSelector)
       return client.page.FilesPageElement.expirationDatePicker()
     },
-    hasCollaboratorsList: async function () {
+    hasCollaboratorsList: async function (expectCollaborator = true) {
       let isVisible = false
       const element = this.elements.collaboratorsList
+
+      const timeout = expectCollaborator ? 5000 : this.api.globals.waitForNegativeConditionTimeout
```
Comment (individual-it) [inline tests/acceptance/pageObjects/FilesPageElement/SharingDialog/collaboratorsDialog.js:64]: please avoid magic numbers

File: tests/acceptance/features/webUIUpload/uploadEdgecases.feature [hunk 4]
```diff
@@ -51,14 +51,16 @@ Feature: File Upload
 
 
   Scenario: overwrite an existing file
+    Given user "Alice" has created file "'single'quotes.txt" in the server
+    And user "Alice" has created file "strängé filename (duplicate #2 &).txt" in the server
+    And user "Alice" has created file "zzzz-must-be-last-file-in-folder.txt" in the server
+    And the user has reloaded the current page of the webUI
```
Comment (individual-it) [inline tests/acceptance/features/webUIUpload/uploadEdgecases.feature:57]: why do we need to reload?

File: tests/acceptance/features/webUIUpload/uploadEdgecases.feature [hunk 5]
```diff
@@ -51,14 +51,16 @@ Feature: File Upload
 
 
   Scenario: overwrite an existing file
+    Given user "Alice" has created file "'single'quotes.txt" in the server
+    And user "Alice" has created file "strängé filename (duplicate #2 &).txt" in the server
+    And user "Alice" has created file "zzzz-must-be-last-file-in-folder.txt" in the server
```
Comment (individual-it) [inline tests/acceptance/features/webUIUpload/uploadEdgecases.feature:56]: how did this work before?

**Review by individual-it — CHANGES_REQUESTED** [review]
generally look good to me, but I would like to avoid the `5000` hardcoded in a lot of places.
I know the magic numbers have been there before, but if we refactor we can get rid of them

File: tests/acceptance/pageObjects/FilesPageElement/SharingDialog/collaboratorsDialog.js (hunk 3 above)
Comment (saw-jan) [reply tests/acceptance/pageObjects/FilesPageElement/SharingDialog/collaboratorsDialog.js:64]: I can use the value from globals

File: tests/acceptance/features/webUIUpload/uploadEdgecases.feature (hunk 5 above)
Comment (saw-jan) [reply tests/acceptance/features/webUIUpload/uploadEdgecases.feature:56]: previously `confirmFileOverwrite` function did fail but the test kept on going.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternal2</strong> https://drone.owncloud.com/owncloud/web/23473/60/1
The following scenarios passed on retry:
- webUISharingInternalUsers/shareWithUsers.feature:342

**Pull request merged**